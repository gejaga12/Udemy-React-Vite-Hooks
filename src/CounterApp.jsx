import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
    // console.log('render');
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    // console.log(counter);
    // setCounter((c) => c + 1); //para aumentar
    setCounter(counter + 1); // para aumentar
  };

  const handleSubtract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <div>
        <h1>CounterApp</h1>
        <p>{counter}</p>
        <button onClick={handleAdd}>+1</button>
        <button onClick={handleSubtract}>-1</button>
        <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
