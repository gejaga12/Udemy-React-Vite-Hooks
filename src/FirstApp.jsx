// import { Fragment } from "react";
// SIN IMPORTAR EL FRAGMENTO UTILIZO <> </> QUE SIMULA UN FRAGMENTO

import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  // console.log(props);
  return (
    <>
      {/* <code>{JSON.stringify(nuevoMsj)}</code> */}
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
    name: PropTypes.string
  };

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subTitle: '123456',
    name: 'Este es mi nombre'
}