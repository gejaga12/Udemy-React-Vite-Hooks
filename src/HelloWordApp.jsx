// function App() {
//     return <h1>Hola React</h1>;
//   }

//   export default App;

// import React from "react";

const HelloWordApp = () => {
  return (
    <div>
      <h1>Hola React</h1>
    </div>
  );
};

export default HelloWordApp;

