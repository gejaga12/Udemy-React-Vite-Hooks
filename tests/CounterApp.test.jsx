import { CounterApp } from "../src/CounterApp";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Pruebas en <CounterApp />", () => {

  const initialValue = 0;

  test("Debe hacer match con el snapshot ", () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el valor inicial de 100 ", () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy(); 
  });

  // test('Debe incremetar con el +1 ', () => {
  //   render(<CounterApp value={initialValue} />);
  //   fireEvent.click( screen.getByText('+1'))
  //   expect(screen.getByText('0')).toBeTruthy();
  // })
  
  // test('Debe incremetar con el -1 ', () => {
  //   render(<CounterApp value={initialValue} />);
  //   fireEvent.click( screen.getByText('-1'))
  //   expect(screen.getByText('-1')).toBeTruthy();
  // })
  
  test('debe de funcionar el botÃ³n de reset', () => {
        
    render( <CounterApp value={ initialValue } /> );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('Reset') );
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

    expect( screen.getByText( 0 ) ).toBeTruthy();
5
});



});
