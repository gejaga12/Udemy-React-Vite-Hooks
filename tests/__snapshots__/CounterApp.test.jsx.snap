// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe hacer match con el snapshot  1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <p>
      0
    </p>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
